contracts/helpers/MultiBlockRebalancer.sol

```diff
@@ -0,0 +1,313 @@
+            // These checks assume that there are no transaction fees on the stablecoin->collateral conversion and so
+            // it's still possible that exposure goes above the max exposure in some rare cases
+            if (stablecoinsFromAsset * 1e9 > maxValueScaled) amount = 0;
+            else if ((stablecoinsFromAsset + amount) * 1e9 > maxValueScaled)
+                amount = maxValueScaled / 1e9 - stablecoinsFromAsset;
+        } else {
+            // In this case, exposure after the operation might remain slightly below the targetExposure as less
+            // collateral may be obtained by burning stablecoins for the yield asset and unwrapping it
+            amount = targetExposureScaled / 1e9 - stablecoinsFromCollateral;
+            uint256 minValueScaled = collatInfo.minExposureYieldAsset * stablecoinsIssued;
+            if (stablecoinsFromAsset * 1e9 < minValueScaled) amount = 0;
+            else if (stablecoinsFromAsset * 1e9 < minValueScaled + amount * 1e9)
+                amount = stablecoinsFromAsset - minValueScaled / 1e9;
+        }
+    }
+
+    function _rebalance(uint8 typeAction, address collateral, uint256 amount) internal {
+        if (amount > maxMintAmount) revert TooBigAmountIn();
+        agToken.mint(address(this), amount);
+        _adjustAllowance(address(agToken), address(transmuter), amount);
+        if (typeAction == 1) {
+            address depositAddress = collateralToDepositAddress[collateral];
+
+            if (collateral == XEVT) {
+                uint256 amountOut = transmuter.swapExactInput(
+                    amount,
+                    0,
+                    address(agToken),
+                    EURC,
+                    address(this),
+                    block.timestamp
+                );
+                _checkSlippage(amount, amountOut, collateral, depositAddress);
+                _adjustAllowance(collateral, address(depositAddress), amountOut);
+                (uint256 shares, ) = IPool(depositAddress).deposit(amountOut, address(this));
+                _adjustAllowance(collateral, address(transmuter), shares);
+                amountOut = transmuter.swapExactInput(
+                    shares,
+                    0,
+                    collateral,
+                    address(agToken),
+                    address(this),
+                    block.timestamp
+                );
+                agToken.burnSelf(amountOut, address(this));
+            } else if (collateral == USDM) {
+                uint256 amountOut = transmuter.swapExactInput(
+                    amount,
+                    0,
+                    address(agToken),
+                    USDC,
+                    address(this),
+                    block.timestamp
+                );
+                _checkSlippage(amount, amountOut, collateral, depositAddress);
+                IERC20(collateral).transfer(depositAddress, amountOut);
+            }
```

Comment:
We can just use the logic for a vault here, as same interface as ERC4626

Reply:
I don't really understand why as USDM is an ERC20 rebasing token